scripts/o3de/o3de/manifest.py

```diff
@@ -319,10 +319,15 @@ def get_project_templates(project_path: pathlib.Path) -> list:
 
 # gem.json queries
 def get_gem_gems(gem_path: pathlib.Path) -> list:
-    return get_gems_from_external_subdirectories(get_gem_external_subdirectories(gem_path))
+    return get_gems_from_external_subdirectories(get_gem_external_subdirectories(gem_path, set()))
 
 
-def get_gem_external_subdirectories(gem_path: pathlib.Path) -> list:
+def get_gem_external_subdirectories(gem_path: pathlib.Path, visited_gem_paths: set) -> list:
```

Comment:
Was python reusing the same set associated with `visited_gem_paths` variable across multiple calls and that is why you needed to pass in a fresh set?
I know python uses name bindings and that causes issues such as the following: https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

Reply:
Good point! I did not run into that issue, just chose to require the set param because I thought it might be safer - in part so the recursion call would fail if I accidentally forgot to provide the set.  A default value in this case would probably be fine.

An interesting thing that happened when running the pytests was that the cycle condition check is triggered by one of the tests that provides two of the same paths in `external_subdirectories` - which isn't a cycle, but nice that it prevents traversing the same paths again.